elements/lisk-p2p/test/integration/outbound_shuffling.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { expect } from 'chai';
+import { P2P } from '../../src/index';
+import { wait } from '../utils/helpers';
+import { platform } from 'os';
+
+describe('Outbound peer shuffling', () => {
+	let p2pNodeList: ReadonlyArray<P2P> = [];
+	const NETWORK_START_PORT = 5000;
+	const NETWORK_PEER_COUNT_SHUFFLING = 10;
+	const POPULATOR_INTERVAL_SHUFFLING = 10000;
+	const OUTBOUND_SHUFFLE_INTERVAL = 500;
+
+	before(async () => {
+		sandbox.restore();
+	});
+
+	after(async () => {
+		await Promise.all(
+			p2pNodeList
+				.filter(p2p => p2p.isActive)
+				.map(async p2p => await p2p.stop()),
+		);
+		await wait(1000);
+	});
+
+	beforeEach(async () => {
```

Comment:
Please, have a general look at the order of our hooks. In this particular file, `beforeEach` should be placed before than `after`.

Reply:
IMO both are valid since it's "before all, after all" and "before each, after each". but I can change it

Reply:
The suggestion is to follow always one approach.